utils/s3/client.cc

```diff
@@ -81,21 +82,8 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
     co_await util::skip_entire_stream(in);
 }
 
-static future<> look_for_errors(const http::reply&, input_stream<char>&& in_) {
-    auto in = std::move(in_);
-    auto body = co_await util::read_entire_stream_contiguous(in);
-    auto possible_error = aws::aws_error::parse(std::move(body));
-    if (possible_error.get_error_type() != aws::aws_error_type::OK) {
-        throw std::system_error(std::error_code(EIO, std::generic_category()), possible_error.get_error_message());
-    }
-}
-
-client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag)
-        : _host(std::move(host))
-        , _cfg(std::move(cfg))
-        , _gf(std::move(gf))
-        , _memory(mem)
-{
+client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
+    : _host(std::move(host)), _cfg(std::move(cfg)), _gf(std::move(gf)), _memory(mem), _retry_strategy(std::move(rs)) {
```

Comment:
Please, don't reformat the whole thing "while at it", it's impossible to review such change.
Also the way it was formatted before is preferred in this case

Reply:
reverted